batch-indexing/src/main/java/data/config/IndexingJobConfig.java

```diff
@@ -0,0 +1,125 @@
+        LOGGER.info("Chunk size: {}, Parallelization size: {}", chunkSize, parallelization);
+        return new JobBuilder(BATCH_JOB, jobRepository)
+                .incrementer(new TimestampJobParametersIncrementer())
+                .start(indexingStep)
+                .build();
+    }
+
+    @Bean
+    public Step indexingStep(
+            JobRepository jobRepository,
+            @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
+            RepositoryItemReader<ExecutionRecord> indexingRepositoryItemReader,
+            ItemProcessor<ExecutionRecord, Future<ExecutionRecordDTO>> indexingAsyncItemProcessor,
+            ItemWriter<Future<ExecutionRecordDTO>> executionRecordDTOAsyncItemWriter) {
+
+        return new StepBuilder(STEP_NAME, jobRepository)
+                .<ExecutionRecord, Future<ExecutionRecordDTO>>chunk(chunkSize, transactionManager)
+                .reader(indexingRepositoryItemReader)
+                .processor(indexingAsyncItemProcessor)
+                .writer(executionRecordDTOAsyncItemWriter)
+                .build();
+    }
+
+    @Bean
+    @StepScope
+    public RepositoryItemReader<ExecutionRecord> indexingRepositoryItemReader(
+            ExecutionRecordRepository executionRecordRepository) {
+        final DefaultRepositoryItemReader defaultRepositoryItemReader = new DefaultRepositoryItemReader(executionRecordRepository);
+        defaultRepositoryItemReader.setPageSize(chunkSize);
+        return defaultRepositoryItemReader;
+    }
+
+    @Bean
+    public ItemProcessor<ExecutionRecord, Future<ExecutionRecordDTO>> indexingAsyncItemProcessor(
+            ItemProcessor<ExecutionRecord, ExecutionRecordDTO> indexingItemProcessor,
+            @Qualifier("indexingStepAsyncTaskExecutor") TaskExecutor indexingAsyncTaskExecutor) {
+        AsyncItemProcessor<ExecutionRecord, ExecutionRecordDTO> asyncItemProcessor = new AsyncItemProcessor<>();
+        asyncItemProcessor.setDelegate(indexingItemProcessor);
+        /*TODO:
+         we don't use task executor here since Mongo client, used in eu.europeana.indexing.IndexerImpl,
+         is sync client (doesn't work in case of multithreaded environment). We can consider to migrate it to reactive client;
+         https://www.mongodb.com/docs/drivers/java/sync/current/
+         */
+//        asyncItemProcessor.setTaskExecutor(indexingAsyncTaskExecutor);
+        return asyncItemProcessor;
+    }
+
+    @Bean
+    @ConfigurationProperties(prefix = "indexing")
+    public IndexingProperties previewIndexingProperties() {
+        return new IndexingProperties();
+    }
+
+    @Bean
+    public IndexingSettings indexingSettings(IndexingProperties indexingProperties) {
+        try {
+            return new IndexingSettingsGenerator(indexingProperties).generate();
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to prepare indexing settings", e);
```

Comment:
Do we need to capture and throw here, of can we just directly throw the exceptions from generate()?

Reply:
We can but in this case we have a bit better encapsulation. `IndexingLobConfig` doesn't depend on the `IndexingSettingsGenerator` code.

Reply:
The encapsulation is in fact, or at least that was the intention when we created the indexing library, the IndexingException. The IndexingException represents a wider exception where other exceptions can be derived from it such as SetupRelatedIndexingException.
The fact that the generate() method also throws URISyntaxException seems to create another layer, while I think it should internally be captured and wrapped around an IndexingException(can be any of the derived ones).
We could of course re-wrap here with yet another exception but I think it creates unnecessary overhead since the job here is related to indexing specifically, if setting it up doesn't work then the job cannot proceed anyway.

Reply:
I slightly modified the code